data-hibernate-jpa/src/main/java/io/micronaut/data/hibernate/operations/AbstractHibernateOperations.java

```diff
@@ -400,15 +400,29 @@ public void bindOne(QueryParameterBinding binding, Object value) {
                     Argument<?> argument = preparedQuery.getArguments()[parameterIndex];
                     Class<?> argumentType = argument.getType();
                     if (Collection.class.isAssignableFrom(argumentType)) {
-                        setParameterList(q, parameterName, value == null ? Collections.emptyList() : (Collection<Object>) value, argument.getFirstTypeVariable().orElse(Argument.OBJECT_ARGUMENT));
+                        Collection<Object> paramCollection;
+                        if (value == null) {
+                            paramCollection = Collections.emptyList();
+                        } else if (value instanceof Collection) {
+                            paramCollection = (Collection<Object>) value;
+                        } else {
+                            // This is the case where collection with one element was passed and during binding collection moved to single value element
```

Comment:
This is needed workaround because of this
```
else if (values != null && values.size() == 1) {
            // Bind as one if the expanded collections is just one item
            value = values.iterator().next();
            values = null;
        }
```
in `DefaultBindableParametersStoredQuery` as we are trying to cast single object to the collection. 
Would be good if we can remove that piece of code from `DefaultBindableParametersStoredQuery`, something like that is likely causing issue #1788 but when looking at option to do it found out other MongoDb errors which we can deal in another issue that I will create

Reply:
Changed DefaultBindableParametersStoredQuery which simplifies many things.